infra/modules/database/main.tf

```
@@ -177,49 +106,56 @@ resource "aws_ssm_parameter" "admin_user" {
   type  = "SecureString"
   value = local.admin_user
 }
+################################################################################
+# Parameters for Query Logging
+################################################################################
 
-############################################################################################
-## Backup Configuration
-############################################################################################
-
-resource "aws_backup_plan" "database" {
-  name = "${var.database_name}-backup-plan"
+# For psql query logging, see https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_LogAccess.Concepts.PostgreSQL.html#USER_LogAccess.Concepts.PostgreSQL.Query_Logging
+resource "aws_db_parameter_group" "rds_query_logging_postgresql" {
+  count       = var.database_type == "postgresql" ? 1 : 0
+  name        = "${var.database_name}-${var.database_type}"
+  family      = "postgres14"
+  description = "Default parameter group"
 
-  rule {
-    rule_name         = "${var.database_name}-backup-rule"
-    target_vault_name = "${var.database_name}-vault"
-    schedule          = "cron(0 12 ? * SUN *)"
+  parameter {
+    name  = "log_statement"
+    value = "ddl" # Only logs major database changes (e.g. ALTER TABLE)
   }
 
-  depends_on = [
-    aws_backup_vault.database
-  ]
+  parameter {
+    name  = "log_min_duration_statement"
+    value = "1"
+  }
 }
 
-# KMS Key for the vault
-# This key was created by AWS by default alongside the vault
-data "aws_kms_key" "database" {
-  key_id = "alias/aws/backup"
-}
+# For mysql query logging, see https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_LogAccess.MySQL.LogFileSize.html
+resource "aws_db_parameter_group" "rds_query_logging_mysql" {
+  count       = var.database_type == "mysql" ? 1 : 0
+  name        = "${var.database_name}-${var.database_type}"
+  family      = "mysql8.0"
+  description = "Default parameter group"
 
-# create backup vault
-resource "aws_backup_vault" "database" {
-  name        = "${var.database_name}-vault"
-  kms_key_arn = data.aws_kms_key.database.arn
+  parameter {
+    name  = "general_log"
+    value = "1"
+  }
 }
 
-# create IAM role
-resource "aws_iam_role" "database_backup" {
-  name               = "${var.database_name}-database-backup"
-  assume_role_policy = data.aws_iam_policy_document.database_backup.json
+################################################################################
+# IAM role for enhanced monitoring
+################################################################################
+
+resource "aws_iam_role" "rds_enhanced_monitoring" {
+  name               = "${var.database_name}-rds-enhanced-monitoring-demo"
```

@aplybeah Since you renamed this resource in `database-serverless/main.tf` to `${var.database_name}-rds-serverless-enhanced-monitoring`, can the resource here in `database/main.tf` just be this:

```suggestion
  name               = "${var.database_name}-rds-enhanced-monitoring"
```